### DISCUSSION TITLE
Running Moddable on a WT32-ETH01

### DISCUSSION THREAD

**Discussion by bartbutenaers** [phoddie/node-red-mcu#65] [Q&A]
Hi everybody,

I am going to split my original question into pieces, to make it readable for future visitors here.

My original question was:
> I don't see the [WT32-ETH01](https://files.seeedstudio.com/products/102991455/WT32-ETH01_datasheet_V1.1-%20en.pdf) in the list of models supported by Moddable, and of course you cannot support all kind of boards. But do you think it will never be possible to run node-red-mcu on such devices, e.g. due to some technical limitations (regarding ethernet libraries or something alike)?

![image](https://user-images.githubusercontent.com/14224149/209781684-eb94a376-1adf-4ab3-80c6-b34d59f1b694.png)

And @phoddie has [answered](https://github.com/phoddie/node-red-mcu/discussions/51#discussioncomment-4330173) it already briefly:
> At a baseline level, the Moddable SDK works with most ESP32 boards out there by compiling to the generic esp32 platform target (rather than a specific device like esp32/moddable_two. Of course, the generic build only knows about the ESP32 and not about any attached peripherals, such as the Ethernet peripheral here. So, a new build platform would be needed for the W32-ETH01. That looks like it would be mostly about configuring the Ethernet. The Moddable SDK does support Ethernet, as described in a blog post. The Ethernet part is uses , the LAN8720A, is supported by the ESP-IDF so should be straightforward to get working.

It would be very nice if you could describe in simple words what *"a new build platform would be needed for the W32-ETH01"* means.  I assume some advance knowledge is required to get something like that up and running.

The reason that I use those wt32-eth01 boards, is that I use them to measure liquid levels (oil, water, ...).  However those tanks are in my garden, out of my wifi range.  Currently I use them with EspEasy which runs very well.  But would like to use Moddable for this...

Thanks (again)!!
Bart

**Comment by andycarle** [comment]
Hi @bartbutenaers,

Supporting this development board looks reasonably straightforward. The main challenge will be bringing up the Ethernet controller.

I like the idea of having support for an additional Ethernet module included in the Moddable SDK, so I'm happy to take a look at this part and see if I can get it going. I've ordered one for arrival early next week. I'll report back when I've had a chance to work with it a bit.

**Reply by bartbutenaers** [reply]
Hi @andycarle,
You people are very generous, to say the least...
Will create some tutorials afterwards, and hopefully attract some more folks to use Moddable.
That is the least I can do in return...

Thanks!!
Bart

**Reply by phoddie** [reply]
The WT32-ETH01 board is supported in the Moddable SDK now. The platform build target is `esp32/wt32_eth01`. We'll be bringing one to demo with Node-RED MCU Edition at [FOSDEM](https://fosdem.org) in Brussels 🇧🇪 February 4th and 5th. @bartbutenaers, will you be able to come take a look? 🤞

**Comment by bartbutenaers** [comment]
@phoddie,
Ah it is implemented already.  Very nice.  Looking forward to test it!!!
Now it is up to me to find some time to play with it...
Will get back to you with the results of my experiments.

I'm afraid I won't be able to go to Fosdem unfortunately.  Horrible lack of free time lately.  I will already be glad if I get Moddable installed on a wt32-eth01 during next weekend...
But I wish you guys/ladies very good luck with the demo in Brussels!!

**Reply by phoddie** [reply]
@bartbutenaers – Good luck with the wt52-eth01. It works. I've programmed one myself. ;) But... it is a very bare-bones board, which makes programming it a little bit more of an adventure that most boards. It requires an external USB-to-Serial bridge to communicate (presumably you have that already, since you've used these before) and there's a dance to jump pin 0 to get it into programming mode. @andycarle can help as needed.
Too bad you can't make it to FOSDEM, but completely understand about life being busy.
